sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -535,3 +538,75 @@ def _to_generated(self) -> GeneratedConfigurationSnapshot:
             retention_period=self.retention_period,
             tags=self.tags,
         )
+
+
+class ConfigurationSettingPropertiesPaged(PageIterator):
+    """An iterable of ConfigurationSetting properties."""
+
+    etag: str
```

Comment:
Optional[str]?

Reply:
It is `str`. For example, when there's no configuration settings in cloud, we can still get 'Etag' value from response header.

Reply:
The return type of response_headers.get('ETag') is Optional[str]. :)

Reply:
'ETag' is required in `list_configuration_settings` operation and this iterator is only used in this operation, I'll change to get the value via `pop` and throw an error if the value is missing in response headers.